packages/common/src/content/_internal.ts

```diff
@@ -89,8 +82,177 @@ const MAX_SIZE = 5000000;
  * @param requestOptions Hub Request Options (including whether we're in portal)
  * @returns
  */
-export const isProxiedCSV = (item: IItem, requestOptions: IHubRequestOptions) =>
-  !requestOptions.isPortal &&
+export const isProxiedCSV = (
+  item: IItem,
+  requestOptions?: IHubRequestOptions
+) =>
+  !isPortal(requestOptions) &&
   item.access === "public" &&
   item.type === "CSV" &&
-  item.size <= MAX_SIZE;
+  item.size <= 5000000;
+
+export const getHubRelativeUrl = (
+  type: string,
+  identifier: string,
+  typeKeywords?: string[]
+): string => {
+  // solution types have their own logic
+  let contentUrl = getSolutionUrl(type, identifier, typeKeywords);
+  if (!contentUrl) {
+    const family = getFamily(type);
+    const familiesWithPluralizedRoute = [
+      "app",
+      "dataset",
+      "document",
+      "map",
+      "template",
+    ];
+    // default to the catchall content route
+    let path = "/content";
+    if (family === "feedback") {
+      // the exception
+      path = "/feedback/surveys";
+    } else if (isPageType(type)) {
+      // pages are in the document family,
+      // but instead of showing the page's metadata on /documents/about
+      // but we render the page on the pages route
+      path = "/pages";
+    } else if (familiesWithPluralizedRoute.indexOf(family) > -1) {
+      // the rule: route name is plural of family name
+      path = `/${family}s`;
+    }
+    contentUrl = `${path}/${identifier}`;
+  }
+  return contentUrl;
+};
+
+export const isPageType = (type: string) =>
+  ["Hub Page", "Site Page"].includes(type);
+
+const getSolutionUrl = (
+  type: string,
+  identifier: string,
+  typeKeywords?: string[]
+): string => {
+  let hubUrl;
+  if (type === "Solution") {
+    // NOTE: as per the above spreadsheet,
+    // solution types are now in the Template family
+    // but we don't send them to the route for initiative templates
+    if (typeKeywords?.indexOf("Deployed") > 0) {
+      // deployed solutions go to the generic content route
+      hubUrl = `/content/${identifier}`;
+    }
+    // others go to the solution about route
+    hubUrl = `/templates/${identifier}/about`;
+  }
+  return hubUrl;
+};
+
+// metadata
+export interface IMetadataPaths {
+  updateFrequency: string;
+  metadataUpdateFrequency: string;
+  metadataUpdatedDate: string;
+  reviseDate: string;
+  pubDate: string;
+  createDate: string;
+}
+
+export function getMetadataPath(identifier: keyof IMetadataPaths) {
+  // NOTE: i have verified that this will work regardless of the "Metadata Style" set on the org
```

Comment:
This is likely because it's the ArcGIS metadata format - but we should verify and document that.